docs/section21/Rule21-1.md

```diff
@@ -0,0 +1,45 @@
+
+# Boiler - Rule 21-1  
+
+**Rule ID:** 21-1  
+**Rule Description:** For systems using purchased hot water or steam, the heating source shall be modeled as purchased hot water or steam in both the proposed design and baseline building design. Hot-water or steam costs shall be based on actual utility rates.  
+**Rule Assertion:** B-RMR ASHRAE229:conditioning_component(external_fluid_source).type = P-RMR ASHRAE229:conditioning_component(external_fluid_source).type  
+**Appendix G Section:** Section 21 Boiler  
+**Appendix G Section Reference:** Section G3.1.1.1 & G3.1.1.3.1 Building System-Specific Modeling Requirements for the Baseline model  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  
+
+1. P-RMR is modeled with purchased hot water or steam
+
+**Manual Check:** None  
+**Evaluation Context:** Building  
+**Data Lookup:** None  
+**Function Call:** None  
+
+## Rule Logic:  
+
+- For each conditioning component in P_RMR: `for conditioning_component_p in P_RMR.ASHRAE229.conditioning_components:`
+
```

Comment:
@xingcx ASHRAE.229 data group does not have a data element for conditioning_components. We could instead check if ASHRAE229.external_fluid_source is not NULL

Reply:
conditioned_component was an old element when this RDS was create. This rule has been updated.